### PR TITLE
harmless cleanup

### PR DESCRIPTION
**Pull request by sundevour**
what's different?

• cute new `isD1` and `isD2` bools to avoid repeatedly comparing destinyVersion and because it's tidy

• adding D1, D2, and `reviewsEnabled` content inline,
for `categoryHashFilters`, `stats`, and `filterTrans`
instead of `.push`ing or `.assign`ing them later.
this keeps the data close together, so there's not multiple locations to check,
and makes the logic extremely spatially clear, what's being added to what and why

• replaces "operand" with "operator". an operator operates on operands

• tunes quote-matching regex with 1 fewer `|`'s, for less need to branch repeatedly while parsing

• renames 2 instances of things just named "filter" so their purpose is slightly clearer

• instead of 44 combined `startWith`s, and `replace`s and a long `if else` chain,
parses the search term once with regex and uses `switch` to check the results.
falls through a bunch on predicates that needed the same stuff done

• some comments!!

### CODE REVIEW

File: src/app/search/search-filters.ts [hunk 1]
```diff
@@ -533,130 +490,119 @@ function searchFilters(
...
       query = query.replace(/[\u2018|\u2019|\u201A]/g, "'");
       query = query.replace(/[\u201C|\u201D|\u201E]/g, '"');
-      // could probably tidy this regex, just a quick hack to support multi term:
-      // [^\s]*?"[^"]+?" -> match is:"stuff here"
-      // [^\s]*?'[^']+?' -> match is:'stuff here'
-      // [^\s"']+' -> match is:stuff
-      const searchTerms = query.match(/[^\s]*?"[^"]+?"|[^\s]*?'[^']+?'|[^\s"']+/g) || [];
+      // \S*?(["']).*?\1 -> match `is:"stuff here"` or `is:'stuff here'`
+      // [^\s"']+ -> match is:stuff
+      const searchTerms = query.match(/\S*?(["']).*?\1|[^\s"']+/g) || [];
```
Comment (bhollis) [inline src/app/search/search-filters.ts]: This is one place where I wish we had unit tests - this regex has been really tricky to get right. I'll trust that you've tested this appropriately, but some issues we've had before includes quotes inside quotes.

File: src/app/search/search-filters.ts [hunk 2]
```diff
@@ -1326,6 +1271,7 @@ function searchFilters(
           item.sockets.sockets.some((socket) => {
             return !!(
               socket.plug &&
+              // these hashes are all forms of "empty socket"
               ![2323986101, 2600899007, 1835369552, 3851138800, 791435474].includes(
```
Comment (bhollis) [inline src/app/search/search-filters.ts]: It'd be nice to pull this into a named constant.

File: src/app/search/search-filters.ts (hunk 1 above)
Comment (sundevour) [reply src/app/search/search-filters.ts]: this accounts for being able to search `"` by using '`'s, or vice versa, and parsed my test cases of multiple quotes etc. it is nothing fancy just the existing regex with the same greed/laziness, but with a backreference

File: src/app/search/search-filters.ts (hunk 2 above)
Comment (sundevour) [reply src/app/search/search-filters.ts]: on it

Comment (delphiactual) [reply src/app/search/search-filters.ts]: it think these all have the following `plug.onActionRecreateSelf` = false

which normal mods do not

Comment (delphiactual) [reply src/app/search/search-filters.ts]: or even `plug.insertionMaterialRequirementHash === 0`

Comment (sundevour) [reply src/app/search/search-filters.ts]: @delphiactual on my list. gonna test & implement it after this, so i can stop battling merge conflicts

File: src/app/search/search-filters.ts [hunk 3]
```diff
@@ -56,53 +94,39 @@ export const searchFiltersConfigSelector = createSelector(
...
-// * loadouts
-// * current character
-// * all items (for dupes)
-// * itemInfo
-// * ratings
-// * newItems
-// * and maybe some other stuff?
+/** A selector for a predicate function for searching items, given the current search query. */
+// TODO: this also needs to depend on: settings / loadouts / current character
+//  all items (for dupes) / itemInfo / ratings / newItems / and maybe some other stuff?
```
Comment (bhollis) [inline src/app/search/search-filters.ts]: We can get rid of this TODO - I actually already did all that.

File: src/app/search/search-filters.ts [hunk 4]
```diff
@@ -331,12 +308,16 @@ export function buildSearchConfig(destinyVersion: 1 | 2): SearchConfig {
 
   return {
     keywordToFilter,
-    keywords: [...new Set(keywords)],
+    keywords: [...new Set(keywords)], // de-dupe kinetic (dmg) & kinetic (slot)
```
Comment (bhollis) [inline src/app/search/search-filters.ts]: We should probably just rename one of them...

**Pull request merged**